modules/core/test/test_intrin_utils.hpp

```diff
@@ -2029,9 +2030,9 @@ void test_hal_intrin_float16()
 {
     DUMP_ENTRY(v_float16);
 #if CV_FP16
-    TheTest<v_float32>()
-        .test_loadstore_fp16_f32()
+    TheTest<v_float32>().test_loadstore_fp16_f32();
 #if CV_SIMD_FP16
+    TheTest<v_float16>()
```

Comment:
Story behind these changes:

1. https://github.com/opencv/opencv/pull/12120 shows that there was only two tests on `v_float16` in the beginning:  `TheTest<v_float16>().test_loadstore_fp16().test_float_cvt_fp16();`
2. Then https://github.com/opencv/opencv/pull/12256 added `TheTest<v_float32>().test_loadstore_fp16_f32();` besides the above test.
3. Now problem came with https://github.com/opencv/opencv/pull/15494, which removed `TheTest<v_float16>()` and attached its belonged tests to `TheTest<v_float32>()` instead. These changes caused problems in test `test_loadstore_fp16()` and `test_float_cvt_fp16()`. They are supposed to test load and store FP16 numbers and FP16<->FP32 conversions if I am not mistaking, so they should not be attached to `TheTest<v_float32>()`.